packages/apps/ability-uniswap-swap/src/lib/ability-helpers/get-uniswap-quote.ts

```diff
@@ -2,128 +2,100 @@ import { Token, CurrencyAmount, TradeType, Percent } from '@uniswap/sdk-core';
 import { AlphaRouter, SwapType, SwapRoute } from '@uniswap/smart-order-router';
 import { ethers } from 'ethers';
 
-export const getUniswapQuote = async ({
+export async function getUniswapQuote({
   rpcUrl,
-  chainId,
   tokenInAddress,
-  tokenInDecimals,
   tokenInAmount,
   tokenOutAddress,
-  tokenOutDecimals,
+  recipient,
+  slippageTolerance,
 }: {
   rpcUrl: string;
-  chainId: number;
   tokenInAddress: string;
-  tokenInDecimals: number;
-  tokenInAmount: number;
+  tokenInAmount: string;
   tokenOutAddress: string;
-  tokenOutDecimals: number;
-}): Promise<{
-  bestQuote: ethers.BigNumber;
-  bestFee: number;
-  amountOutMin: ethers.BigNumber;
-  route: SwapRoute;
-}> => {
-  console.log('Getting Uniswap Quote (getUniswapQuote)', {
-    rpcUrl,
-    chainId,
-    tokenInAddress,
-    tokenInDecimals,
-    tokenInAmount,
-    tokenOutAddress,
-    tokenOutDecimals,
-  });
+  recipient: string;
+  slippageTolerance?: number;
+}): Promise<SwapRoute> {
+  const activeTimeouts = new Set<NodeJS.Timeout>();
+  const realSetTimeout = global.setTimeout;
+  const realClearTimeout = global.clearTimeout;
 
-  const provider = new ethers.providers.StaticJsonRpcProvider(rpcUrl);
-  const router = new AlphaRouter({ chainId, provider });
+  global.setTimeout = ((fn: (...args: unknown[]) => void, ms?: number, ...args: unknown[]) => {
+    const id = realSetTimeout(fn, ms as number, ...args);
+    activeTimeouts.add(id);
+    return id;
+  }) as typeof setTimeout;
 
-  // Create token instances
-  const tokenIn = new Token(chainId, tokenInAddress, tokenInDecimals);
-  const tokenOut = new Token(chainId, tokenOutAddress, tokenOutDecimals);
+  global.clearTimeout = ((id: NodeJS.Timeout) => {
+    activeTimeouts.delete(id);
+    return realClearTimeout(id);
+  }) as typeof clearTimeout;
```

Comment:
Because this is still be ran within the Uniswap quote generator/signer Lit Action, we require this manual timeout clean up logic to keep the Lit Action from timing out due to hanging timers